src/aap_eda/analytics/utils.py

```diff
@@ -31,65 +30,85 @@ def datetime_hook(dt: dict) -> dict:
-            verify = str_to_bool(verify)
-
-        token = inputs.get("oauth_token")
-
-        controller_info["credential_id"] = credential.id
-        controller_info["inputs"] = inputs
-        if token:
-            headers = {"Authorization": f"Bearer {token}"}
-            logger.info("Use Bearer token to ping the controller.")
-        else:
-            user_pass = f"{inputs.get('username')}:{inputs.get('password')}"
-            auth_value = (
-                f"Basic {base64.b64encode(user_pass.encode()).decode()}"
-            )
-            headers = {"Authorization": f"{auth_value}"}
-            logger.info("Use Basic authentication to ping the controller.")
 
+    for credential in credentials:
         try:
-            resp = requests.get(url, headers=headers, verify=verify)
-            resp_json = resp.json()
-            controller_info["install_uuid"] = resp_json["install_uuid"]
-
+            inputs = yaml.safe_load(credential.inputs.get_secret_value())
+            host = inputs["host"].removesuffix("/api/controller/")
+            url = f"{host}/api/v2/ping/"
+            verify = inputs.get("verify_ssl", False)
+            token = inputs.get("oauth_token")
+
+            headers = _build_auth_headers(inputs, token)
+            controller_info = {
+                "credential_id": credential.id,
+                "inputs": inputs,
+            }
+
+            # quickly to retrieve controller's info. timeout=3
+            resp = requests.get(url, headers=headers, verify=verify, timeout=3)
+            resp.raise_for_status()
+            controller_info["install_uuid"] = resp.json()["install_uuid"]
             info[host] = controller_info
+
+        except KeyError as e:
+            logger.error(f"Missing key in credential inputs: {e}")
+            continue
+        except yaml.YAMLError as e:
+            logger.error(
+                f"YAML parsing error for credential {credential.id}: {e}"
+            )
+            continue
         except requests.exceptions.RequestException as e:
             logger.warning(
-                "Failed to connect with controller using credential "
-                f"{credential.name}: {e}"
+                f"Controller connection failed for {credential.name}: {e}"
             )
+            continue
+        except Exception as e:
+            logger.error(
+                f"Unexpected error processing credential {credential.id}: {e}"
+            )
+            continue
 
     return info
 
 
+def _build_auth_headers(inputs: dict, token: Optional[str]) -> dict:
+    if token:
+        return {"Authorization": f"Bearer {token}"}
+    else:
+        username = inputs.get("username")
+        password = inputs.get("password")
+        if not (username and password):
+            raise ValueError("Missing username/password for Basic Auth")
+        auth_str = base64.b64encode(f"{username}:{password}".encode()).decode()
+        return {"Authorization": f"Basic {auth_str}"}
```

Comment:
It is safer and more maintainable approach for handling authentication if we already use an existing implementation, `requests.auth.HTTPBasicAuth` In the end you are going to use `requests` for the request, so there is no reason to reinvent the authentication.